ndcube/ndcollection.py

```diff
@@ -0,0 +1,262 @@
+                item_strings = [isinstance(_item, str) for _item in item]
+                item_is_strings = all(item_strings)
+                # Ensure strings are not mixed with slices.
+                if (not item_is_strings) and (not all(np.invert(item_strings))):
+                    raise TypeError("Cannot mix keys and non-keys when indexing instance.")
+
+            # If sequence is all strings, extract the cubes corresponding to the string keys.
+            if item_is_strings:
+                new_data = [self[_item] for _item in item]
+                new_keys = item
+                new_aligned_axes = tuple([self.aligned_axes[_item] for _item in item])
+
+            # Else, the item is assumed to be a typical slicing item.
+            # Slice each cube in collection using information in this item.
+            # However, this can only be done if there are aligned axes.
+            else:
+                if self.aligned_axes is None:
+                    raise IndexError("Cannot slice unless collection has aligned axes.")
+                else:
+                    # Derive item to be applied to each cube in collection and
+                    # whether any aligned axes are dropped by the slicing.
+                    collection_items, new_aligned_axes = self._generate_collection_getitems(item)
+                    # Apply those slice items to each cube in collection.
+                    new_data = [self[key][tuple(cube_item)]
+                                for key, cube_item in zip(self, collection_items)]
+                    # Since item is not strings, no cube in collection is dropped.
+                    # Therefore the collection keys remain unchanged.
+                    new_keys = list(self.keys())
+
+            return self.__class__(new_data, keys=new_keys, aligned_axes=new_aligned_axes,
+                                  meta=self.meta, dont_sanitize_aligned_axes=True)
+
+    def _generate_collection_getitems(self, item):
+        # There are 3 supported cases of the slice item: int, slice, tuple of ints and/or slices.
+        # Compile appropriate slice items for each cube in the collection and
+        # and drop any aligned axes that are sliced out.
+
+        # First, define empty lists of slice items to be applied to each cube in collection.
+        collection_items = [[slice(None)] * len(self[key].dimensions) for key in self]
+        # Define empty list to hold aligned axes dropped by the slicing.
+        drop_aligned_axes_indices = []
+
+        # Case 1: int
+        # First aligned axis is dropped.
+        if isinstance(item, int):
+            drop_aligned_axes_indices = [0]
+            # Insert item to each cube's slice item.
+            for i, key in enumerate(self):
+                collection_items[i][self.aligned_axes[key][0]] = item
+
+        # Case 2: slice
+        # If only one element in interval of slice, first aligned axis is dropped.
```

Comment:
is this true? A length 1 slice doesn't drop a dimension in a numpy array:
```
$ a = np.random.random([10, 10])                                                                                                                                                                                                                               
$ a[0:1].shape                                                                                                                                                                                                                                                 
(1, 10)
```

Reply:
You're absolutely right.  This was an old comment that needs updating.  The expected behaviour of ```NDCollection``` should be consistent with your example above.